src/panels/config/helpers/dialog-helper-detail.ts

```diff
@@ -0,0 +1,170 @@
+              ) || this._platform
+            )
+          : this.hass.localize("ui.panel.config.helpers.dialog.add_helper")}
+      >
+        ${this._platform
+          ? html`
+              <div class="form" @value-changed=${this._valueChanged}>
+                ${this._error
+                  ? html`
+                      <div class="error">${this._error}</div>
+                    `
+                  : ""}
+                ${dynamicElement(`ha-${this._platform}-form`, {
+                  hass: this.hass,
+                  item: this._item,
+                })}
+              </div>
+              <mwc-button
+                slot="primaryAction"
+                @click="${this._createItem}"
+                .disabled=${this._submitting}
+              >
+                ${this.hass!.localize("ui.panel.config.helpers.dialog.create")}
+              </mwc-button>
+            `
+          : html`
+              <div class="container">
+                ${Object.keys(HELPERS).map((platform: string) => {
+                  return html`
+                    <mwc-button
+                      .disabled=${!isComponentLoaded(this.hass, platform)}
+                      @click="${this._platformPicked}"
+                      .platform="${platform}"
+                    >
+                      ${this.hass.localize(
+                        `ui.panel.config.helpers.types.${platform}`
+                      ) || platform}
+                    </mwc-button>
+                  `;
+                })}
+              </div>
+            `}
+      </ha-dialog>
+    `;
+  }
+
+  private _valueChanged(ev: CustomEvent): void {
+    this._item = ev.detail.value;
+  }
+
+  private async _createItem(): Promise<void> {
+    if (!this._platform || !this._item) {
+      return;
+    }
+    this._submitting = true;
+    this._error = "";
+    try {
+      await HELPERS[this._platform](this.hass, this._item);
+      this.closeDialog();
+    } catch (err) {
+      this._error = err.message || "Unknown error";
+    } finally {
+      this._submitting = false;
+    }
+  }
+
+  private _platformPicked(ev: Event): void {
+    this._platform = (ev.currentTarget! as any).platform;
+  }
+
+  static get styles(): CSSResult {
+    return css`
+      ha-dialog {
+        --mdc-dialog-title-ink-color: var(--primary-text-color);
+        --justify-action-buttons: space-between;
+      }
+      @media only screen and (min-width: 600px) {
+        ha-dialog {
+          --mdc-dialog-min-width: 600px;
+        }
+      }
+
+      /* make dialog fullscreen on small screens */
+      @media all and (max-width: 450px), all and (max-height: 500px) {
+        ha-dialog {
+          --mdc-dialog-min-width: 100vw;
+          --mdc-dialog-max-height: 100vh;
+          --mdc-dialog-shape-radius: 0px;
+          --vertial-align-dialog: flex-end;
+        }
+      }
+      .error {
+        color: var(--google-red-500);
+      }
+      .container {
+        display: grid;
```

Comment:
I don't think that a grid is easy to process, I prefer a list. List is fine as we only have  couple of items.